protocol-dashboard/src/hooks/useConnectAudiusProfile.ts

```diff
@@ -1,49 +1,54 @@
+import { OAUTH_URL } from '@audius/sdk'
 import { useQueryClient } from '@tanstack/react-query'
 import { getDashboardWalletUserQueryKey } from 'hooks/useDashboardWalletUsers'
 import { useEffect } from 'react'
 import { audiusSdk } from 'services/Audius/sdk'
-import { useUser } from 'store/cache/user/hooks'
-import {
-  AUDIUS_NETWORK_ID,
-  ETH_NETWORK_ID,
-  disableRefreshAfterNetworkChange,
-  switchNetwork
-} from 'utils/switchNetwork'
 
 const env = import.meta.env.VITE_ENVIRONMENT
 
+let resolveUserHandle = null
+let receiveUserHandlePromise = null
+
+const receiveUserId = async (event: MessageEvent) => {
+  const sdk = await audiusSdk()
+  const oauthOrigin = new URL(OAUTH_URL[env]).origin
+  if (
+    event.origin !== oauthOrigin ||
+    event.source !== sdk.oauth.activePopupWindow ||
+    !event.data.state
+  ) {
+    return
+  }
+  if (sdk.oauth.getCsrfToken() !== event.data.state) {
+    console.error('State mismatch.')
+    return
+  }
+  if (event.data.userHandle != null) {
+    resolveUserHandle(event.data.userHandle)
+  }
```

Comment:
Should we maybe have a reject case here so the hook doesn't stall forever in the case where userHandle is null?